pkgs/tools/typesetting/soupault/default.nix

```diff
@@ -0,0 +1,39 @@
+{ fetchgit, ocamlPackages, lib }:
+
+ocamlPackages.buildDunePackage rec {
+  pname = "soupault";
+  version = "3.1.0";
+
+  useDune2 = true;
+
+  src = fetchgit {
+    url = "https://github.com/dmbaturin/soupault.git";
+    rev = "${version}";
+    sha256 = "sha256-SVNC2DbdciunSKTCmmX0SqaEXMe1DkVX4VJTqriI8Y4=";
+  };
```

Comment:
```suggestion
  src = fetchFromGitHub {
    owner = "dmbaturin";
	 repo = pname;
    rev = version;
    sha256 = "sha256-SVNC2DbdciunSKTCmmX0SqaEXMe1DkVX4VJTqriI8Y4=";
  };
```

Reply:
Why though? There's a precedence about changing this to a different Git repo in the future if more people support the getting away. Why should GitHub get special treatment? `fetchgit` is Git-forge-agnostic and would create a 1-line diff if that changed instead of a 5-line diff.

Reply:
Because `fetchFromGitHub` is faster (it downloads a tarball) and is *always* used when a github repository is available.
If the repo changes its git URL then the package will just have to be updated to reflect that. If only one line is changed, or 4 doesn't matter

And github doesn't have special treatment, the same function exists for sourcehut, gitlab, and a bunch of other similar sites - list of most of them is here: https://nixos.org/manual/nixpkgs/stable/#chap-pkgs-fetchers

Reply:
okay... i'll amend